src/command_modules/azure-cli-security/azure/cli/command_modules/security/custom.py

```diff
@@ -0,0 +1,455 @@
+    else:
+        tasks = client.tasks.list()
+
+    return tasks
+
+
+def get_security_task(
+        client,
+        resource_name,
+        resource_group_name=None):
+
+    for loc in client.locations.list():
+        client.config.asc_location = loc.name
+
+    if resource_group_name:
+        task = client.tasks.get_resource_group_level_task(resource_group_name, resource_name)
+    else:
+        task = client.tasks.get_subscription_level_task(resource_name)
+
+    return task
+
+
+# --------------------------------------------------------------------------------------------
+# Security Alerts
+# --------------------------------------------------------------------------------------------
+
+def list_security_alerts(
+        client,
+        resource_group_name=None,
+        location=None):
+
+    if location:
+        client.config.asc_location = location
+
+        if resource_group_name:
+            alerts = client.list_resource_group_level_alerts_by_region(resource_group_name)
+        else:
+            alerts = client.list_subscription_level_alerts_by_region()
+
+    elif resource_group_name:
+        alerts = client.list_by_resource_group(resource_group_name)
+    else:
+        alerts = client.list()
+
+    return alerts
+
+
+def get_security_alert(
+        client,
+        location,
+        resource_name,
+        resource_group_name=None):
+
+    client.config.asc_location = location
+
+    if resource_group_name:
+        alert = client.get_resource_group_level_alerts(resource_name, resource_group_name)
+    else:
+        alert = client.get_subscription_level_alert(resource_name)
+
+    return alert
+
+
+def dismiss_security_alert(
+        client,
+        location,
+        resource_name,
+        resource_group_name=None):
+
+    action = 'Dismiss'
+    client.config.asc_location = location
+
+    if resource_group_name:
+        client.update_resource_group_level_alert_state(resource_name, action, resource_group_name)
+    else:
+        client.update_subscription_level_alert_state(resource_name, action)
+
+
+def activate_security_alert(
+        client,
+        location,
+        resource_name,
+        resource_group_name=None):
+
+    action = 'Activate'
+    client.config.asc_location = location
+
+    if resource_group_name:
+        client.update_resource_group_level_alert_state(resource_name, action, resource_group_name)
+    else:
+        client.update_subscription_level_alert_state(resource_name, action)
+
+# --------------------------------------------------------------------------------------------
+# Security Settings
+# --------------------------------------------------------------------------------------------
+
+
+def list_security_settings(
+        client):
+
+    settings = client.list()
+
+    return settings
+
+
+def get_security_setting(
+        client,
+        resource_name):
+
+    setting = client.get(resource_name)
+
+    return setting
```

Comment:
Please condense all these functions, you can put the arguments on the same line and you do not need to save the result to a variable if you are just returning it.

Reply:
changed